regserver/handlers/search.go

```diff
@@ -20,9 +24,20 @@ func NewSearchHandler(s registry.Searchable) http.HandlerFunc {
 				return
 			}
 		default:
-			err := fmt.Errorf("Content-Type must be application/json")
-			apiutil.WriteErrResponse(w, http.StatusBadRequest, err)
-			return
+			// read form values
+			limit, err := apiutil.ReqParamInt("limit", r)
```

Comment:
these assignments should be refactored:
```golang
var err error
if p.Limit, err = apiutil.ReqParamInt("limit", r); err != nil {
    p.Limit = defaultLimit
}
```
can do the same for offset, but `var err error` should only be declared once. Also, note the `=` instead of `:=` above